pipeline_manager/src/main.rs

```diff
@@ -434,13 +560,20 @@ struct UpdateProjectResponse {
 /// Change project code and/or name.
 ///
 /// If project code changes, any ongoing compilation gets cancelled,
-/// project status is reset to `ProjectStatus::None`, and project version
+/// project status is reset to `None`, and project version
 /// is incremented by 1.  Changing project name only doesn't affect its
 /// version or the compilation process.
-///
-/// # HTTP errors
-///
-/// * `NOT_FOUND` - `project_id` does not exist in the database.
+#[utoipa::path(
```

Comment:
these don't seem to save you much typing.
Are they mostly for documentation using swagger?

Reply:
utoipa generates an OpenAPI spec, which is a piece of JSON that other tools can use to generate client-side bindings, e.g., I'm working on Python bindings now.  And it also generates swagger documentation from that.